pkg/container/client.go

```diff
@@ -398,8 +441,33 @@ func (cl *Client) GetManifest(ctx context.Context, digest digest.Digest, local b
 	}
 
 	if err = retry.RetryIfNecessary(ctx, func() error {
-		r.Data, r.MimeType, err = src.GetManifest(ctx, nil)
-		return err
+		var secondaryDigest *digest.Digest
+		// passing in a digest for a local manifest list will break
+		if instanceDigest != "" && !local {
```

Comment:
Thanks for the comment here, I am struggling a bit here and wonder if we can somehow tweak it to make it clearer to me but probably followup material.

Reply:
Yeah it might also be a naming thing. It was a bit tricky because this function is called recursively, so if we are getting this info for a single image then it will be empty but if we are calling the function on a manifest list we pass in the child digest. If that makes sense?

Reply:
Yeah, probably followup material at this point (sorry, I really don't want slow things down here). I did play a little bit and was thinking maye:
```go
index d2476f656..42e0b140e 100644
--- a/pkg/container/client.go
+++ b/pkg/container/client.go
@@ -396,15 +396,26 @@ func (cl *Client) getLocalImageIDFromDigest(instance digest.Digest) (string, err
 // GetManifest fetches the raw manifest data from the server. If digest is not empty
 // it will override any given tag for the Client's Target.
 func (cl *Client) GetManifest(ctx context.Context, instanceDigest digest.Digest, local bool) (r RawManifest, err error) {
-       var id string
-       if instanceDigest != "" && local {
-               id, err = cl.getLocalImageIDFromDigest(instanceDigest)
-               if err != nil {
-                       return r, err
+       var localID string
+       var overrideDigest *digest.Digest
+
+       if instanceDigest != "" {
+               if local {
+                       localID, err = cl.getLocalImageIDFromDigest(instanceDigest)
+                       if err != nil {
+                               return r, err
+                       }
+               } else {
+                       // We can pass the instance digest, if it is nil, then this is the primary manifest.
+                       // If it is not nil, then this is the instance digest and the primary manifest is a
+                       // manifest list. The `GetManifest` call will then retrieve the manifest for the
+                       // desired instance, see:
+                       // https://github.com/containers/image/blob/cdb2f596a95018444f4dee0993e321d3d8bc328d/types/types.go#L252C2-L253C121
+                       overrideDigest = &instanceDigest
                }
        }
 
-       ref, err := cl.getImageRef(id, local)
+       ref, err := cl.getImageRef(localID, local)
        if err != nil {
                return
        }
@@ -425,17 +436,7 @@ func (cl *Client) GetManifest(ctx context.Context, instanceDigest digest.Digest,
        }
 
        if err = retry.RetryIfNecessary(ctx, func() error {
-               var secondaryDigest *digest.Digest
-               // passing in a digest for a local manifest list will break
-               if instanceDigest != "" && !local {
-                       // We can pass the instance digest, if it is nil, then this is the primary manifest.
-                       // If it is not nil, then this is the instance digest and the primary manifest is a
-                       // manifest list. The `GetManifest` call will then retrieve the manifest for the
-                       // desired instance, see:
-                       // https://github.com/containers/image/blob/cdb2f596a95018444f4dee0993e321d3d8bc328d/types/types.go#L252C2-L253C121
-                       secondaryDigest = &instanceDigest
-               }
-               r.Data, r.MimeType, err = src.GetManifest(ctx, secondaryDigest)
+               r.Data, r.MimeType, err = src.GetManifest(ctx, overrideDigest)
                if err != nil {
                        return err
                }
```
would help but I'm not sure. WDYT? I will try to mediate a bit more later (but again, not a blocker, this is really just me)

Reply:
Yeah I can have a think about it too. But I love the detail and the quality of the review. Thanks again for this. I'll get this in and then start adding some of this to the follow up